private/buf/cmd/buf/command/push/push.go

```diff
@@ -267,6 +268,40 @@ func validateCreateFlags(flags *flags) error {
 }
 
 func validateLabelFlags(flags *flags) error {
+	if err := validateLabelFlagCombinations(flags); err != nil {
+		return err
+	}
+	return validateLabelFlagValues(flags)
+}
+
+// We need to validate the combination of flags that set labels:
+//   - do not allow mixing of `--label` and `--tag` flags
+//   - do not allow mixing of `--label` and `--branch`/`--draft` flags
+//   - do not allow mixing of `--tag` and `--branch`/`--draft` flags
+//   - do not allow mixing of `--draft` and `--branch` flags
+func validateLabelFlagCombinations(flags *flags) error {
```

Comment:
Nit: because there are multiple combinations here, what about this approach?

```go
usedFlags := make(map[string]struct{})
if len(flags.Labels) > 0 {
  usedFlags[labelFlagName] = struct{}
}
if len(flags.Tags) > 0 {
  usedFlags[tagFlagName] = struct{}
}
if flags.Branch != "" {
  usedFlags[branchFlagName] = struct{}
}
if flags.Draft != "" {
  usedFlags[draftFlagName] = struct{}
}
if len(usedFlags) > 1 {
  // compose an error message with all the flags that are used
}
```